web/ui/react-app/src/App.tsx

```diff
@@ -7,9 +7,19 @@ import {
   Row,
 } from 'reactstrap';
 
+import './App.css';
 import PanelList from './PanelList';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faSearch, faSpinner, faCheck } from '@fortawesome/free-solid-svg-icons';
+import { faSquare, faCheckSquare } from '@fortawesome/free-regular-svg-icons';
 
-import './App.css';
+library.add(
```

Comment:
There's already a couple of `library.add()` statements in the app, but they are local to the components that need them (`TimeInput`, `ExpressionInput`, and `GraphControls`). Wouldn't it make sense to keep that pattern where every component adds what it needs? I also see this call adding some icons that other `library.add()` calls add (and which the checkbox doesn't need), but without removing those other calls?

Reply:
So, as per library docs this should be placed in the root Component in order icons to be available in the whole app and it make sense if same icons are used in multiple places. Thus this kind of setup will be in one place but generally, I would suggest finding another, more simple solution instead of library.add(), dom.watch(), importing more than one package, and so on :) WDYT?

Reply:
Do you really need to use the `library.add()`? 
I was under the impression that you can get tree shaking (if that's what you're after) by importing each icon manually and passing it "raw" with:

```
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrash } from "@fortawesome/free-solid-svg-icons/faTrash";
[...]
<FontAwesomeIcon icon={faTrash} />
```

Reply:
Yes, this pattern definitely looks better and no additional setup is needed :)

Reply:
That seems even better, yeah :)

I bet when I did this back in February (while learning TypeScript + React at the same time), I just saw a page like https://fontawesome.com/how-to-use/with-the-api/setup/library and assumed that one should use the library.